src/fileops/SwiftIO.cpp

```diff
@@ -0,0 +1,250 @@
+            break; // EOF
+        }
+
+        if(remaining == 0) {
+            DAVIX_SLOG(DAVIX_LOG_DEBUG, DAVIX_LOG_CHAIN, "Data provider buffer has been filled");
+            break; // buffer is full
+        }
+    }
+
+    DAVIX_SLOG(DAVIX_LOG_DEBUG, DAVIX_LOG_CHAIN, "Retrieved {} bytes from data provider", written);
+    return written;
+}
+
+std::string SwiftIO::writeChunk(IOChainContext &iocontext, const char *buff, dav_size_t size, int partNumber) {
+    Uri url(iocontext._uri);
+    url.setPath(url.getPath() + "/" + std::to_string(partNumber));
+
+    DAVIX_SLOG(DAVIX_LOG_DEBUG, DAVIX_LOG_CHAIN, "writing chunk #{} with size {}", partNumber, size);
+
+    DavixError * tmp_err=NULL;
+    PutRequest req(iocontext._context, url, &tmp_err);
+    checkDavixError(&tmp_err);
+
+    req.setParameters(iocontext._reqparams);
+    req.setRequestBody(buff, size);
+    req.executeRequest(&tmp_err);
+    if(!tmp_err && httpcodeIsValid(req.getRequestCode()) == false){
+        httpcodeToDavixError(req.getRequestCode(), davix_scope_io_buff(),
+                             "write error: ", &tmp_err);
+    }
+    DAVIX_SLOG(DAVIX_LOG_DEBUG, DAVIX_LOG_CHAIN, "write result size {}", size);
+    checkDavixError(&tmp_err);
+
+    std::string etag;
+    if(!req.getAnswerHeader("Etag", etag)) {
+        DavixError::setupError(&tmp_err, "Swift::MultiPart", StatusCode::InvalidServerResponse, "Unable to retrieve chunk Etag, necessary when committing chunks");
+    }
+
+    DAVIX_SLOG(DAVIX_LOG_DEBUG, DAVIX_LOG_CHAIN, "chunk #{} written successfully, etag: {}", partNumber, etag);
+    return etag;
+}
+
+void SwiftIO::commitInlineChunks(IOChainContext & iocontext, const std::vector<Prop> &props, const size_t MaxManifestSegments){
+    Uri url(iocontext._uri);
+
+    size_t iterNum = 0;
+    size_t remaining = props.size();
+    size_t propBegin = 1;
+    size_t propEnd = MaxManifestSegments;
+    std::string lastEtag;
+    std::string path = url.getPath();
+
+    while(remaining > 0){
+        // generate a multipart manifest in json
+        std::ostringstream manifest;
+        manifest << "[";
+        if(iterNum > 0){
+            manifest << "{";
+            manifest << "\"path\":\"" << path << "-" << iterNum - 1 << "\",";
+            manifest << "\"etag\":" << lastEtag << "},";
+        }
+        for(size_t i = propBegin; i <= propEnd; i++) {
+            manifest << "{";
+            manifest << "\"path\":\"" << path << "/" << i << "\",";
+            manifest << "\"etag\":\"" << props[i - 1].first << "\",";
+            manifest << "\"size_bytes\":" << props[i - 1].second << "}";
+            if(i != propEnd) {
+                manifest << ',';
+            }
+        }
+        manifest << "]";
+
+        remaining -= propEnd - propBegin + 1;
+        propBegin = propEnd + 1;
+        propEnd = std::min(propEnd + MaxManifestSegments - 1, propEnd + remaining);
```

Comment:
Shouldn't it be `propEnd + MaxManifestSegments`? 
(as you're increasing it by exactly one interval  `= MaxManifestSegments`)